pkg/pfconfig/shared/shared.go

```diff
@@ -108,10 +107,90 @@ func (c SX1301Config) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m)
 }
 
-// UnmarshalJSON implements json.Unmarshaler. It currently just errors because
-// unmarshaling is not supported.
-func (c *SX1301Config) UnmarshalJSON(_ []byte) error {
-	return errUnmarshalNotImplemented
+var (
+	radioKey = regexp.MustCompile("^radio_([0|1])$")
+	txLutKey = regexp.MustCompile("^tx_lut_([0-9]{1,2})$")
+	chanKey  = regexp.MustCompile("^chan_multiSF_([0-9]{1,2})$")
+)
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (c *SX1301Config) UnmarshalJSON(msg []byte) error {
+	var root map[string]json.RawMessage
+	if err := json.Unmarshal(msg, &root); err != nil {
+		return err
+	}
+	radioMap, txLutMap, chanMap := make(map[int]RFConfig), make(map[int]TxLUTConfig), make(map[int]IFConfig)
+	for key, value := range root {
+		if key == "lorawan_public" {
```

Comment:
Better to use a `switch` statement here, no need to do all comparisons for every key.

```go
switch {
	case key == "lorawan_public":
	// ...
}
```